### PR TITLE
Update Repo to 3.0

### PR DESCRIPTION
**Pull request by ksr30**
## Changes
- Updated Depnedecies to 3.0.
- Added new functionality introduced by 3.0 like Trait -> Config. 
- Added what I suggested in #85.
- Updated Mock.rs of all pallets to make all test cases working.

### CODE REVIEW

**Comment by ksr30** [comment]
Hey @ansermino please have look. And please also note that I updated https://github.com/ksr30/chainbridge-substrate/blob/master/example-pallet/src/lib.rs#L80. (#85 )

**Comment by ansermino** [comment]
Hey @ksr30,

Thanks so much for this! Looks great, just two small things to address before merging:
- I'm unable to produce the error you reported (#85). I'm not opposed to the change but I would like to understand why. What rust version where you using? 
- Please update the github actions workflow to use a more recent nightly compiler version

**Comment by ksr30** [comment]
Hey @ansermino,
I resolved all warnings. And I am using latest toolchain version [rustc 1.52.0-nightly (4f20caa62 2021-03-01)
]. Do you want me to add this version in github actions workflow? Thanks

**Comment by ansermino** [comment]
Hey @ksr30,

Thanks! Yeah if you could bump the version to that it would be much appreciated. Just want to make sure things won't suddenly break sometime in the future 🙂

File: example-pallet/src/lib.rs
Comment (ansermino) [inline example-pallet/src/lib.rs:80]: I'm not convinced this will always work. There may be cases where BalanceOf represents an unsigned 256-bit integer. Can you try removing the explicit type and confirm that it builds? Seems to work on my end.

Comment (ksr30) [reply example-pallet/src/lib.rs]: Hey @ansermino 
yes, its working with nightly-2020-10-06 but throwing error which I mentioned in #85 with '1.52.0-nightly'. 
For now I removed it but I didn't update nightly version.

Comment (ansermino) [reply example-pallet/src/lib.rs]: You're right. I must have mixed up my compiler versions. I believe the problem still exists that we want to support 256bit numbers and this is limiting us to 128bits. Any ideas?

Comment (ksr30) [reply example-pallet/src/lib.rs]: I think [amount](https://github.com/ChainSafe/chainbridge-substrate/blob/main/example-pallet/src/lib.rs#L18) is already of type u128. Because If you see your runtime :- [Balance](https://github.com/ChainSafe/chainbridge-substrate-chain/blob/master/runtime/src/lib.rs#L57) is of type u128. 
See these links also :-
https://github.com/ChainSafe/chainbridge-substrate-chain/blob/master/runtime/src/lib.rs#L291
https://github.com/ChainSafe/chainbridge-substrate-chain/blob/master/runtime/src/lib.rs#L236
https://github.com/ChainSafe/chainbridge-substrate-chain/blob/master/runtime/src/lib.rs#L57

So basically this `U256::from(amount.saturated_into())`  is converting from u128 to u256.

Comment (ansermino) [reply example-pallet/src/lib.rs]: You are right, but I was hoping this could be written in a way to be added to any runtime. By utilizing traits & generics we should be able to make this less restricted. 

Please add the conversion back in and update GH actions to use the more recent compiler. I don't love this solution but this shouldn't block this PR from being merged. Thanks for your patience 🙏

Comment (ksr30) [reply example-pallet/src/lib.rs]: Hey @ansermino, can you please create an issue for that or provide more detail for that. I will try to look into it whenever I get time. Meanwhile, for this PR, I am going to do what you said.

Comment (ansermino) [reply example-pallet/src/lib.rs:80]: Thanks for your patience! Will do 😄

**Review by ansermino — APPROVED** [review]

**Pull request merged**
